### PR TITLE
Remove InplaceStringBuilder

### PR DESCRIPTION
**Pull request by G-flat**
Resolves issue #35994.

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39163) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39163) before we can accept your contribution.<br/><br/>:x: G-flat [sign now](https://cla.dotnetfoundation.org/dotnet/runtime?pullRequest=39163)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/runtime?pullRequest=39163) it.</sub>

**Comment by davidfowl** [comment]
RIP

**Comment by PriyaPurkayastha** [comment]
Please create a breaking change issue either using https://github.com/dotnet/docs/issues/new?template=dotnet-breaking-change.md or an ASP.NET Core breaking change announcement issue using https://github.com/aspnet/Announcements/issues/new?template=breaking-change.md (as mentioned in https://github.com/dotnet/runtime/issues/35994) so that it gets onto the radar of our content development team for documentation.

File: src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs [hunk 1]
```diff
@@ -219,7 +219,7 @@ static string GetJoinedStringValueFromArray(string[] values)
                         length += value.Length;
                     }
                 }
-#if NETCOREAPP || NETSTANDARD2_1
+
                 // Create the new string
                 return string.Create(length, values, (span, strings) => {
```
Comment (maryamariyan) [inline src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs:224]: I don't think you can remove the #ifdef here.

string.Create was not introduced for netstandard2.0 and build this project for netstandard2.0 also.

Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.Primitives/src/StringValues.cs:224]: As per @eerhardt 's recommendation in other PR you could perhaps try using ValueStringBuilder and keep the condition here. Something like: https://github.com/dotnet/runtime/commit/f07a1d9564f238b3928113d21bfbc43216bc62df

**Comment by eerhardt** [comment]
@G-flat - the non-net5.0 builds are failing because we are using a method that doesn't exist on old platforms. Will you be able to make the changes need for CI to pass?

If not, let us know and we can help out here.

**Comment by maryamariyan** [comment]
@G-flat thanks again for the PR. Since we are approaching code complete for 5.0 issues, let us know if you'd be open to updating this PR with the feedback this week before the submission window closes.

**Comment by maryamariyan** [comment]
Closing as dupe of https://github.com/dotnet/runtime/pull/40743.

**Pull request closed without merging**